### PR TITLE
Vmi/upload file

### PR DESCRIPTION
**Pull request by Nikeleos**

### CODE REVIEW

File: public/locales/en/translation.json [hunk 1]
```diff
@@ -206,9 +214,7 @@
       "text" : {
         "scenariotype" : "Run type"
       }
-    }
-  },
-  "genericcomponent": {
+    },
```
Comment (csm-thu) [inline public/locales/en/translation.json:217]: This change seems to break some labels (e.g. the "loading lines" labels). Same thing in the FR dict.

File: src/components/ScenarioParameters/ScenarioParameters.js [hunk 2]
```diff
@@ -175,31 +185,46 @@ const ScenarioParameters = ({
...
   const handleClickOnLaunchScenarioButton = () => {
-    // If scenario parameters have never been updated, do it now
-    if (!currentScenario.data.parametersValues) {
-      handleClickOnUpdateAndLaunchScenarioButton();
-    } else {
-      launchScenario(workspaceId, scenarioId);
-      changeEditMode(false);
-    }
+    launchScenario(workspaceId, scenarioId);
+    changeEditMode(false);
```
Comment (csm-thu) [inline src/components/ScenarioParameters/ScenarioParameters.js:192]: This change may break the current behavior when clicking on "launch" just after a scenario creation, unless the parametersValues have been set by the back-end. Was it done on purpose ?

File: src/components/ScenarioParameters/UploadFileUtils.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+const getValueFromParameters = (parameters, parameterToSelect) => {
+  if (parameters.current === null) {
+    return parameterToSelect.defaultValue;
+  }
+  const param = parameters.current?.find(element => element.parameterId === parameterToSelect.id);
+  if (param !== undefined) {
+    return param.value;
+  }
+  return parameterToSelect.defaultValue;
+};
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/components/ScenarioParameters/UploadFileUtils.js:28]: :+1:

File: src/components/ScenarioParameters/ScenarioParameters.js [hunk 4]
```diff
@@ -45,124 +55,124 @@ const ScenarioParameters = ({
...
+        currencyValueParam,
+        currencyUsedParam,
+        startDateValueParam
       ]);
     }
-
-    // TODO Add file upload parameters if necessary
-    // TODO Add array template parameters if necessary
+    /// //////////////////////////////////////////////////////////////////////// INITIAL STOCK
+    // TODO Add initial stock parameters correctly!!!!
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/components/ScenarioParameters/ScenarioParameters.js:160]: :+1:

File: src/components/ScenarioParameters/ScenarioParameters.js (hunk 2 above)
Comment (jreynard-code) [reply src/components/ScenarioParameters/ScenarioParameters.js:192]: :+1:

File: src/components/ScenarioParameters/UploadFileConfig.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+export const STORAGE_ROOT_DIR_PLACEHOLDER = '%WORKSPACE_FILE%';
+
+// Initial stock configuration
+export const INITIAL_STOCK_PARAM_ID = 'initial_stock_dataset';
+export const AZURE_STORAGE_CONNECTOR_ID = 'C-XPv4LBVGAL';
+
+export const ACCEPT_FILE_TYPE_FILE_UPLOAD =
+    '.xlsx,.xls,.csv,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
+
+export const MIME_TYPE_FILE_UPLOAD = 'application/json';
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/components/ScenarioParameters/UploadFileConfig.js:14]: I renamed respectively AZURE_STORAGE_CONNECTOR_ID and ACCEPT_FILE_TYPE_FILE_UPLOAD into INITIAL_STOCK_PARAM_CONNECTOR_ID and INITIAL_STOCK_PARAM_ACCEPT_FILE_TYPE in order to show that these params are scenario parameter specific.

File: src/components/ScenarioParameters/ScenarioParameters.js [hunk 6]
```diff
@@ -45,124 +55,124 @@ const ScenarioParameters = ({
...
+    setRestockQuantity(UploadFileUtils.getValueFromParameters(defaultScenarioParameters, RESTOCK_PARAM));
+    setWaitersNumber(UploadFileUtils.getValueFromParameters(defaultScenarioParameters, NBWAITERS_PARAM));
+
+    // Basic Types Sample
+    setCurrency(UploadFileUtils.getValueFromParameters(defaultScenarioParameters, CURRENCY_PARAM));
+    setCurrencyName(UploadFileUtils.getValueFromParameters(defaultScenarioParameters, CURRENCY_NAME_PARAM));
+    setCurrencyValue(UploadFileUtils.getValueFromParameters(defaultScenarioParameters, CURRENCY_VALUE_PARAM));
+    setCurrencyUsed(UploadFileUtils.getValueFromParameters(defaultScenarioParameters, CURRENCY_USED_PARAM));
+    setStartDate(UploadFileUtils.getValueFromParameters(defaultScenarioParameters, START_DATE_PARAM));
+    /// //////////////////////////////////////////////////////////////////////// INITIAL STOCK
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/components/ScenarioParameters/ScenarioParameters.js:123]: These lines are just here to mark the code to modify is you want to change the information set in the default upload file tab sample

**Review by COSMO-VMI — APPROVED** [review]
LGTM, GG

**Pull request merged**